blocks/bricks/base.py

```diff
@@ -331,14 +337,18 @@ def __getattr__(self, name):
         # These always belong to the parent (the unbound application)
         if name in ('delegate_function', 'properties'):
             return getattr(self.application, name)
+        if name in self.properties.values():
+            return getattr(self.application.brick, name)
         if name in self.properties:
-            return self.properties[name](self.brick)
+            return getattr(self, self.properties[name])(self.brick)
         # First try the parent (i.e. class level), before trying the delegate
         try:
             return getattr(self.application, name)
         except AttributeError:
             if self.delegate_function:
-                return getattr(self.delegate_function(self.brick), name)
+                return getattr(getattr(self.brick,
+                                       self.delegate_function)(),
```

Comment:
`delegate_function` in this case is `'compute_states_delegate'`, so this part here retrieves the method `compute_states_delegate` and calls it, which returns `self.transition.apply`. It then performs `getattr(self.transition.apply, name)`. Nothing changed from before, it's just that `delegate_function` used to be an object, the unbound method `compute_states_delegate`, now it is the name of the method which needs to be retrieved from the class first (and because we retrieve it from the class instance, it is returned bounded already, which is why we don't need to pass `self.brick` as an argument anymore).